### PR TITLE
Added \ilRandom()->int()

### PR DESCRIPTION
**Pull request by smeyer-ilias**
As mentioned in https://mantis.ilias.de/view.php?id=26392 the usage of php rand() does not deliver reliable random numbers for PHP versions < 7.1 (since PHP 7.1 rand() is an alias for mt_rand(), which delivers better random numbers). 

This PR introduces a wrapper class for random number generation and replaces usages of rand(). 
Please check whether the changes in your modules/services. 

If all agree I will merge this PR in 5.4 and 6 and trunk. The version_compare part in class.ilRandom will be deleted.

### CODE REVIEW

**Comment by alex40724** [comment]
Changes in my components are fine. Thanks.

File: Services/Randomization/service.xml
Comment (mjansenDatabay) [inline Services/Randomization/service.xml:3]: A new `ilCtrl` structure reload is req. for this.

**Comment by chfsx** [comment]
thanks for the changes, LGTM!

File: Services/Randomization/service.xml
Comment (smeyer-ilias) [reply Services/Randomization/service.xml:3]: Yes, I know. Wanted to bypass conflicts until this PR is approved.

**Comment by mbecker-databay** [comment]
100% fine overengineering for T&A and WFE purposes where no cryptographical safety of rand is required (as per files in the changeset) but the PR should work on those ends.

**Review by mjansenDatabay — APPROVED** [review]
Approved if the missing database update step will be added.

**Comment by mjansenDatabay** [comment]
@swiniker / @Amstutz Coud you please check the changes regarding the WebDav/Styles components and give @smeyer-ilias feedback? Thx in advance.

**Review by kergomard — APPROVED** [review]
Change in WebDAV LGTM.

**Comment by mjansenDatabay** [comment]
@Amstutz Any objections here regarding the changes in `Styles` https://github.com/ILIAS-eLearning/ILIAS/pull/2429/files#diff-1fb1effe5aa8149cf997c50107d81a7fR1786 ?

**Comment by mjansenDatabay** [comment]
@smeyer-ilias Could you please resolve the conflict in `lang/ilias_de.lang`.

Hi @Amstutz , the TB regularly monitors the list of PRs for old ones. Coud you please review this PR and approve or reject?

**Comment by mjansenDatabay** [comment]
@alex40724 Could you please review this in regards of `Services/Style` and `Media Objects`

**Comment by Amstutz** [comment]
sry for missing out on this, clearly my bad for not responsing. The change in styles is in the content style section not system styles. So I forward this to @akill .

**Review by alex40724 — APPROVED** [review]

**Pull request merged**
